homeassistant/components/motion_blinds/__init__.py

```diff
@@ -0,0 +1,97 @@
+"""The motion_blinds component."""
+from asyncio import TimeoutError as AsyncioTimeoutError
+from datetime import timedelta
+import logging
+from socket import timeout
+
+from homeassistant import config_entries, core
+from homeassistant.const import CONF_API_KEY, CONF_HOST
+from homeassistant.exceptions import ConfigEntryNotReady
+from homeassistant.helpers import device_registry as dr
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator
+
+from .const import DOMAIN, KEY_COORDINATOR, KEY_GATEWAY, MANUFACTURER
+from .gateway import ConnectMotionGateway
+
+_LOGGER = logging.getLogger(__name__)
+
+MOTION_PLATFORMS = ["cover", "sensor"]
+
+
+async def async_setup(hass: core.HomeAssistant, config: dict):
+    """Set up the Motion Blinds component."""
+    return True
+
+
+async def async_setup_entry(
+    hass: core.HomeAssistant, entry: config_entries.ConfigEntry
+):
+    """Set up the motion_blinds components from a config entry."""
+    hass.data.setdefault(DOMAIN, {})
+    host = entry.data[CONF_HOST]
+    key = entry.data[CONF_API_KEY]
+
+    # Connect to motion gateway
+    connect_gateway_class = ConnectMotionGateway(hass)
+    if not await connect_gateway_class.async_connect_gateway(host, key):
+        raise ConfigEntryNotReady
+    motion_gateway = connect_gateway_class.gateway_device
+
+    async def async_update_data():
+        """Fetch data from the gateway and blinds."""
+        try:
+            await hass.async_add_executor_job(motion_gateway.Update)
+            for blind in motion_gateway.device_list.values():
+                await hass.async_add_executor_job(blind.Update)
```

Comment:
Please avoid multiple jumps into the executor.  Make a new function and call it once with `hass.async_add_executor_job`

Reply:
Done